code/modules/cooking/machinery/smartfridge.dm

```diff
@@ -409,6 +400,8 @@
 	return data
 
 /obj/machinery/smartfridge/Topic(href, href_list)
+	if(stat & (NOPOWER|BROKEN) || !anchored)
```

Comment:
This stat check is already handled by the parent via `/obj/machinery/CanUseTopic`

Reply:
Yeah, should have checked the parent given how jank smartfridge code is. This actually made me look over how I was handling it as a whole, and now it follows how vending machines did it, with unanchoring actually giving it the NOPOWER flag, and using power_change() to update.

Works better with these changes, thank you.